src/chain-adapters/NEAR/NEAR.ts

```diff
@@ -0,0 +1,122 @@
+  Transaction as NearTransaction,
+  SignedTransaction as NearSignedTransaction,
+  createTransaction as nearCreateTransaction,
+  encodeTransaction as nearEncodeTransaction,
+  Action as NearAction,
+  Transfer as NearTransfer,
+  Signature as NearSignature,
+} from '@near-js/transactions'
+import { PublicKey as NearPublicKey } from '@near-js/crypto'
+import { baseDecode } from '@near-js/utils'
+import { JsonRpcProvider } from '@near-js/providers'
+
+// Cross-runtime SHA-256 (browser & Node)
+const sha256Bytes = async (data: Uint8Array): Promise<Uint8Array> => {
+  const cryptoAny = (globalThis as any).crypto
+  if (cryptoAny && cryptoAny.subtle) {
+    const digest = await cryptoAny.subtle.digest('SHA-256', data)
+    return new Uint8Array(digest)
+  }
+  const { createHash } = await import('node:crypto')
+  return new Uint8Array(createHash('sha256').update(data).digest())
+}
+
+import type { ChainSignatureContract } from '@contracts/ChainSignatureContract'
+import type { HashToSign, Signature, RSVSignature } from '@types'
+import type { Transaction as SolanaTransaction } from '@solana/web3.js'
+import { ChainAdapter } from '@chain-adapters/ChainAdapter'
+
+import type { NearTransactionRequest, NearUnsignedTransaction } from './types'
+
+export class NEAR extends ChainAdapter<NearTransactionRequest, NearUnsignedTransaction> {
+  private readonly provider: JsonRpcProvider
+  private readonly contract: ChainSignatureContract
+  private readonly networkId: 'mainnet' | 'testnet'
+
+  constructor(args: { rpcUrl: string; networkId: 'mainnet' | 'testnet'; contract: ChainSignatureContract }) {
+    super()
+    this.provider = new JsonRpcProvider({ url: args.rpcUrl })
+    this.contract = args.contract
+    this.networkId = args.networkId
+  }
+
+  private isAccountDoesNotExistError(error: unknown): boolean {
+    const msg = (error as any)?.message?.toString?.() || ''
+    const type = (error as any)?.type?.toString?.() || ''
+    return type === 'AccountDoesNotExist' || msg.includes('AccountDoesNotExist') || msg.includes("doesn't exist") || msg.includes('does not exist')
+  }
+
+  async getBalance(address: string): Promise<{ balance: bigint; decimals: number }> {
+    try {
+      const res = (await this.provider.query<any>(`account/${address}`, '')) as any
+      return { balance: BigInt(res.amount), decimals: 24 }
+    } catch (e) {
+      if (this.isAccountDoesNotExistError(e)) {
+        throw new Error(
+          `NEAR derived account not found: ${address}. Create & fund it or call chainAdapters.near.utils.ensureDerivedAccountExists(...) before sending.`
+        )
+      }
+      throw e
+    }
+  }
+
+  async deriveAddressAndPublicKey(predecessor: string, path: string): Promise<{ address: string; publicKey: string }> {
+    const derivedKey = await this.contract.getDerivedPublicKey({ path, predecessor, IsEd25519: true })
+    const pk = NearPublicKey.fromString(derivedKey as string)
+    const derivedAccountId = `${path}.${predecessor}`
```

Comment:
this look weird to mee, since I would have expected the derived account to be: 
- either an `implicit account`
- either `${predecessor}` (always return `predecessor` indifferent to the `path`)